branch-manager.plugin.zsh

```diff
@@ -78,3 +78,41 @@ function merge_branch {
   echo "✓ Succesfully merged $other_branch into $current_branch"
   echo "$reset_color"
 }
+
+function rebase_branch {
+  current_branch=$(git symbolic-ref --short HEAD)
+  stashed_changes=$(git stash)
+  gitdir="$(git rev-parse --git-dir)"
+  hook="$gitdir/hooks/post-checkout"
+
+  # Rebase from master if no argument given
+  [[ -z "$1" ]] && other_branch="master" || other_branch=$1
+
+  # disable post-checkout hook temporarily
+  [ -x $hook ] && chmod -x $hook
+
+  # Update the requested branch
+  echo "Updating $other_branch…\n"
+  git checkout $other_branch
+  git pull
+
+  # Return to current branch
+  git checkout $current_branch
+
+  # Re-enable hook
+  chmod +x $hook
+
+  # Merge changes
+  git rebase $other_branch
```

Comment:
what happens here on an unclean rebase?  This plus the stash pop might get weird.

Reply:
I personally don’t use rebase, so @blimmer I’ll rely on you to investigate that edge case. Thanks for the review @onyxraven!

Reply:
Good call out @onyxraven - it's actually the same thing that occurs with the `merge_branch` command. It will prevent the `stash pop` from occurring and make you resolve the rebase.

```
 /tmp/example   foobar ●  rebase_branch master
Updating master…

Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
Already up-to-date.
Switched to branch 'foobar'
chmod: .git/hooks/post-checkout: No such file or directory
First, rewinding head to replay your work on top of it...
Applying: whoa nelly
Using index info to reconstruct a base tree...
M	.gitignore
Falling back to patching base and 3-way merge...
Auto-merging .gitignore
CONFLICT (content): Merge conflict in .gitignore
error: Failed to merge in the changes.
Patch failed at 0001 whoa nelly
The copy of the patch that failed is found in: .git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

.gitignore: needs merge
unable to refresh index

✓ Succesfully rebased foobar onto master
```